packages/grafana-ui/src/components/Link/TextLink.tsx

```diff
@@ -0,0 +1,116 @@
+import { css } from '@emotion/css';
+import React, { AnchorHTMLAttributes, forwardRef } from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+
+import { GrafanaTheme2, locationUtil, textUtil, ThemeTypographyVariantTypes } from '@grafana/data';
+
+import { useTheme2 } from '../../themes';
+import { IconName } from '../../types';
+import { Icon } from '../Icon/Icon';
+
+export interface Props extends AnchorHTMLAttributes<HTMLAnchorElement> {
+  /** url to which redirect the user, external or internal */
+  href: string;
+  /** Color to use for text */
+  color?: keyof GrafanaTheme2['colors']['text'] | 'error' | 'success' | 'warning' | 'info';
+  /** If the link will redirect users to a page in or out Grafana */
+  externalLink?: boolean;
+  /** Specify the icon name to be used */
+  icon?: IconName;
+  /** To clarify if the link is a block of if it is part of a bigger text */
```

Comment:
I would put the true statement first, and make it really clear what it does, "True if the element should be displayed inline with surrounding text, false if it should be displayed as a block"